### DISCUSSION TITLE
Where is flatten?

### DISCUSSION THREAD

**Discussion by UlrichVonRekkenin** [davisking/dlib#2988] [Q&A] [answered]
Where is a `flatten` aka in TF?

**Comment by arrufat** [comment, marked as answer]
If you're talking about that functionality in the dnn part, there isn't because it's not needed. Usually, you use flatten in TF to input the tensor into a Fully Connected (or Dense, as TF calls it) layer. In dlib you can input the tensor without flattening, and it will still work. See any `dnn_` example.

Flattening is a human construct, the data remains the same, we just need to connect all neurons in FC layers, no matter where they are in space.

**Reply by UlrichVonRekkenin** [reply]
great, that's what I understood from the code
